src/app/icd/ICDConfig.h

```diff
@@ -0,0 +1,22 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#if CHIP_HAVE_CONFIG_H
```

Comment:
why this #if ?

Reply:
Didn't find the explanation but all the *BuildConfig have the same structure so kept it inline.

Reply:
resolving in favor of this discussion
https://github.com/project-chip/connectedhomeip/pull/30661#discussion_r1406453986